components/dashboard/src/components/delete-account-view.tsx

```diff
@@ -34,87 +34,83 @@ export class DeleteAccountView extends React.Component<DeleteAccountViewProps, D
+                                </ul>
+                            </div>
+                            <div>
+                                Please type in your username <b>{this.props.user.name}</b> to confirm: <br />
+                                <TextField value={this.state.enteredUsername} onChange={e => this.setState({ enteredUsername: (e.target as HTMLInputElement).value })} />
+                            </div>
+                        </Typography>
+                    </DialogContent>
+                    <DialogActions>
+                        <Button onClick={() => this.setState({ status: undefined })} variant="outlined" color="primary">Cancel</Button>
+                        <Button
+                            onClick={() => this.deleteAccount()} variant="outlined"
+                            color="secondary" disabled={this.props.user.name !== this.state.enteredUsername}>
+                            I'm Sure. Delete My Account!
                 </Button>
-            </DialogActions>
-            </React.Fragment>}
+                    </DialogActions>
+                </React.Fragment>}
 
-            {this.state.status == "in-progress" && <React.Fragment>
-            <DialogTitle>We are sorry to see you go.</DialogTitle>
-            <DialogContent>
-                <CircularProgress />
-            </DialogContent>
-            </React.Fragment>}
+                {this.state.status === "in-progress" && <React.Fragment>
+                    <DialogTitle>We are sorry to see you go.</DialogTitle>
+                    <DialogContent>
+                        <CircularProgress />
+                    </DialogContent>
+                </React.Fragment>}
 
-            {this.state.status == "done" && <React.Fragment>
-            <DialogTitle>We are sorry to see you go.</DialogTitle>
-            <DialogContent>
-                <Typography variant="body1">
-                    <div>Your account has been deleted &mdash; all that's left is to log out using the finish button below.</div>
-                    <div>If you ever decide to give Gitpod another try, just sign up like you did the first time. No hard feelings 🙂</div>
-                </Typography>
-            </DialogContent>
-            <DialogActions>
-                <Button onClick={() => window.location.href=new GitpodHostUrl(window.location.toString()).withApi({pathname: '/logout'}).toString()} color="secondary" variant="outlined">Finish</Button>
-            </DialogActions>
-            </React.Fragment>}
+                {this.state.status === "done" && <React.Fragment>
+                    <DialogTitle>We are sorry to see you go.</DialogTitle>
+                    <DialogContent>
+                        <Typography variant="body1">
+                            <div>Your account has been deleted and your browser will be redirected to the logout.</div>
+                            <div>Feel free to try out Gitpod again. No hard feelings 🙂</div>
```

Comment:
@gtsiolis @svenefftinge Is the messaging ok here?

Reply:
BTW, 'reworded, because "to logout using the finish button below" is part of the problem solve in this PR.

Reply:
**question**: Do we need this extra (second) dialog for technical reasons?
- If not, I'd suggest removing it and redirecting to `www.gitpod.io` after confirming deletion in the first dialog. In the future we could redirect to a dedicated the sign up page.
- If yes, here are some potential improvements that could go into this PR:
   - _..will be redirected to the logout_ does not sound accurate, right? We could skip mentioning the redirection.
   - We could rename the second sentence to something like _Thanks for trying out Gitpod_.

**issue**: This is out of the scope of this PR, but visible page contents during page reload are also an issue for anonymous users, see  https://github.com/gitpod-io/gitpod/issues/2509.

Thanks for the ping @csweichel! 🏓

Reply:
@gtsiolis, thanks! 

great hint, and there isn't a technical reason anymore (as of this PR.)

follow up on this suggestion: would you expect to see any confirmation of the successful deletion at all, or is a plain redirection to `www.gitpod.io` just right?

Reply:
@gtsiolis, please give it a try 🙏🏻

Reply:
@AlexTugarev looks good! I don't think we will need the second dialog in the long run but a loading indicator makes sense. Long term we could either use a loading button state (see screenshot below) or a page-wide progress bar like the one we use for environment (workspace) loading.

<img width="160" alt="Screenshot 2021-01-12 at 3 48 38 PM (2)" src="https://user-images.githubusercontent.com/120486/104322695-eae83180-54ed-11eb-90db-e6f3fafb4f55.png">

Once we have a dedicated the sign up page we can consider redirecting users there so that both gitpod.io and self hosted users have a similar overall UX.

Reply:
> I don't think we will need the second dialog 

which one? 

maybe it's the structure of the React component which is a bit confusing? there are three ways to render a `We are sorry to see you go` dialog. I mean that's all the same dialog, but the content is changing to indicate progress and resolution.

Reply:
@AlexTugarev I'm referring to the second dialog shown after confirming deletion that contains the title _We are sorry to see you go_ and the body message _Redirecting..._.

What I meant earlier is that we do not need the second dialog at all. But leaving this as is for now is fine.

> but the content is changing to indicate progress and resolution.

For this, see https://github.com/gitpod-io/gitpod/pull/2692#discussion_r555786711.

Reply:
I see. I didn't even notice the last rendering of `Redirecting...`, maybe because the redirect happens to be fast enough. All three dialog states were a single dialog for me. The idea was to fill in the gap between dismissing the progress indicator and a potentially slow redirect of the browser.

>  But leaving this as is for now is fine.
Can you then give a thumb up on merging?